src/CoreWCF.WebHttp/src/CoreWCF/UriTemplate.cs

```diff
@@ -0,0 +1,1799 @@
+            result.BaseUri = baseUri;
+            if (uriQuery != null)
+            {
+                result.SetQueryParameters(uriQuery);
+            }
+
+            result.SetRelativePathSegments(relativePathSegments);
+            result.Data = data;
+            result.Template = this;
+            for (int i = 0; i < numMatchedSegments; i++)
+            {
+                _segments[i].Lookup(result.RelativePathSegments[i], result.BoundVariables);
+            }
+
+            if (_wildcard != null)
+            {
+                _wildcard.Lookup(numMatchedSegments, result.RelativePathSegments,
+                    result.BoundVariables);
+            }
+            else if (numMatchedSegments < _segments.Count)
+            {
+                BindTerminalDefaults(numMatchedSegments, result.BoundVariables);
+            }
+
+            if (_queries.Count > 0)
+            {
+                foreach (KeyValuePair<string, UriTemplateQueryValue> kvp in _queries)
+                {
+                    kvp.Value.Lookup(result.QueryParameters[kvp.Key], result.BoundVariables);
+                    //UriTemplateHelpers.AssertCanonical(varName);
+                }
+            }
+
+            if (_additionalDefaults != null)
+            {
+                foreach (KeyValuePair<string, string> kvp in _additionalDefaults)
+                {
+                    result.BoundVariables.Add(kvp.Key, UnescapeDefaultValue(kvp.Value));
+                }
+            }
+
+            Fx.Assert(result.RelativePathSegments.Count - numMatchedSegments >= 0, "bad segment computation");
+            result.SetWildcardPathSegmentsStart(numMatchedSegments);
+
+            return result;
+        }
+
+        internal bool IsPathPartiallyEquivalentAt(UriTemplate other, int segmentsCount)
+        {
+            // Refer to the note on template equivalency at IsEquivalentTo
+            // This method checks if any uri with given number of segments, which can be matched
+            //  by this template, can be also matched by the other template.
+            Fx.Assert(segmentsCount >= _firstOptionalSegment - 1, "How can that be? The Trie is constructed that way!");
+            Fx.Assert(segmentsCount <= _segments.Count, "How can that be? The Trie is constructed that way!");
+            Fx.Assert(segmentsCount >= other._firstOptionalSegment - 1, "How can that be? The Trie is constructed that way!");
+            Fx.Assert(segmentsCount <= other._segments.Count, "How can that be? The Trie is constructed that way!");
+            for (int i = 0; i < segmentsCount; ++i)
+            {
+                if (!_segments[i].IsEquivalentTo(other._segments[i],
+                    ((i == segmentsCount - 1) && (IgnoreTrailingSlash || other.IgnoreTrailingSlash))))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+        internal bool IsQueryEquivalent(UriTemplate other)
+        {
```

Comment:
Missing blank line

Reply:
This is addressed in e95ff7c922db5d4a6ee55847b763b49619f08311.